sound/soc/msm/msm8x16.c

```diff
@@ -542,16 +540,18 @@ static int lineout_status_put(struct snd_kcontrol *kcontrol,
 	pr_debug("%s: external speaker PA mode:%d\n", __func__, state);
 
 	switch (state) {
+	case 0:
+		schedule_delayed_work(&lineout_amp_disable, msecs_to_jiffies(50));
+		break;
 	case 1:
 		schedule_delayed_work(&lineout_amp_enable, msecs_to_jiffies(50));
 		break;
-	case 0:
-		//schedule_delayed_work(&lineout_amp_disable, msecs_to_jiffies(5));
-		msm8x16_ext_spk_control(0);
-		break;
 	case 2:
 		schedule_delayed_work(&lineout_amp_dualmode, msecs_to_jiffies(50));
 		break;
+	case 3:
```

Comment:
Special case 3 removed and the noise was expected when downloading from the connected 4-pins headset. And I do not understand the part about https://github.com/Vgdn1942/android_kernel_lenovo_msm8916/blob/cf4430fd40186ad4e83571a1486ab419453ee829/sound/soc/msm/msm8x16.c#L273 if there is assigned a value of state of ucontrol-> value.integer.value [0 ]; https://github.com/Vgdn1942/android_kernel_lenovo_msm8916/blob/cf4430fd40186ad4e83571a1486ab419453ee829/sound/soc/msm/msm8x16.c#L539 ?